.github/workflows/update_changelog.yml

```diff
@@ -0,0 +1,57 @@
+name: Update Changelog.md
+
+on:
+  workflow_run:
+    workflows: ["Check PR for Changelog"]
+    types:
+      - completed
+  pull_request:
+    types:
+      - closed
+
+jobs:
+  update-changelog:
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+    permissions: write-all
+
+    steps:
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Check if PR is from dependabot
+        id: check-dependabot
+        run: |
+          is_dependabot=$(jq -r '.pull_request.user.login' $GITHUB_EVENT_PATH)
+          echo "is_dependabot=$is_dependabot" >> $GITHUB_OUTPUT
+        
+      - name: Extract and update Changelog
+        id: modify-changelog
+        run: |
+          if [ "${{ steps.check-dependabot.outputs.is_dependabot }}" == "dependabot[bot]" ]; then
+          changelog="- **Description:** ${{ github.event.pull_request.title }}\n  - **Products impact:** Dev Dependency upgrade\n  - **Addresses:** -\n  - **Components:** -\n  - **Breaking:** -\n  - **Impacts a11y:** -\n  - **Guidance:** -"
+          else
+            description=$(jq -r ".pull_request.body" "$GITHUB_EVENT_PATH")
+            changelog_section=$(echo "$description" | awk '/## Changelog/{flag=1; next} /##/{flag=0} flag' | sed '/<!--/,/-->/d' | sed '/<!--/,/-->/d' | sed '1d;$d')
+            changelog="${changelog_section}"
+          fi
+
+          pr_number=$(jq --raw-output .pull_request.number "$GITHUB_EVENT_PATH")
+          pr_link="[#${pr_number}]"
+          pr_link_ref="[#${pr_number}]: ${{ github.event.pull_request.html_url }}"
+
+          echo "- ${pr_link}" > pr_info.txt
+          echo -e "  ${changelog}" >> pr_info.txt
+          echo >> pr_info.txt
+          echo "${pr_link_ref}" >> pr_info.txt
+          echo >> pr_info.txt
+
+          sed -i "/<!-- Put all new updates into this section, please -->/r pr_info.txt" CHANGELOG.md
```

Comment:
We will need to use something else to identify the correct place to pasted the new item because "Put all new updates into this section, please" was just a temporary comment. I will think about it more and come back. Perhaps we could add some kind of special comment to the changelog that would be obviously intended for this action to prevent ourselves from deleting it accidentally.

I wonder if it would make sense to actually do the same in our PR template

Reply:
Yeah sure. I can make the changes once we've got that part figured out. We just need a short comment to serve as a marker for the workflow.

Reply:
Thank you, I just posted a suggestion down there

Reply:
One other thing is that this gets a little more complex when we have a release branch _and_ the main branch in play. At the moment, this action will be pushing all updates to the default (main) branch, but we actually want to check out the target branch of the closed PR instead to make the CHANGELOG updates. I see you are taking that into account below, but not for the checkout, so this will attempt to push to the branch that the PR was based on, but with the default branch, so a release branch will get pushed to be `main` instead.